v2/tests/log_context_property_boolean_type_ut/log_context_property_boolean_type_ut.c

```diff
@@ -0,0 +1,466 @@
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_004: [ If any error is encountered, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).to_string shall fail and return a negative value. ]*/
+static void when_snprintf_fails_bool_to_string_also_fails(void)
+{
+    // arrange
+    char buffer[TEST_BUFFER_SIZE];
+    bool bool_value = -1;
+
+    setup_mocks();
+    expected_calls[0].mock_call_type = MOCK_CALL_TYPE_snprintf;
+    expected_calls[0].snprintf_call.call_result = -1;
+    expected_calls[0].snprintf_call.override_result = true;
+    expected_call_count = 1;
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).to_string(&bool_value, buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy */
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_005: [ If src_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall fail and return a non-zero value. ]*/
+static void bool_copy_called_with_NULL_dst_value_fails(void)
+{
+    // arrange
+    bool src = 42;
+
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy(NULL, &src);
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_006: [ If dst_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall fail and return a non-zero value. ]*/
+static void bool_copy_called_with_NULL_src_value_fails(void)
+{
+    // arrange
+    bool dst;
+
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy(&dst, NULL);
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_007: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall copy the bytes of the bool value from the address pointed by src_value to dst_value. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_008: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall succeed and return 0. ]*/
+static void bool_copy_succeeds(void)
+{
+    // arrange
+    bool src = 42;
+    bool dst = 43;
+
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy(&dst, &src);
+
+    // assert
+    POOR_MANS_ASSERT(result == 0);
+    POOR_MANS_ASSERT(dst == 1);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_007: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall copy the bytes of the bool value from the address pointed by src_value to dst_value. ]*/
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_BOOLEAN_TYPE_07_008: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(bool).copy shall succeed and return 0. ]*/
+static void bool_copy_succeeds_2(void)
+{
+    // arrange
+    bool src = INT32_MAX;
+    bool dst = 43;
```

Comment:
Same comment about the values we're testing with. These are clearly copy/pasta from the int32_t. We should have at least 2 the tests exercise the code with true/false.